website/blog/2021-11-22-sql-surrogate-keys.md

```diff
@@ -112,15 +116,59 @@ output:
 | `null`  | 123        | `null`         |
 
 
-You can get around this by wrapping each of your columns in a `coalesce` function to default nulls to blank (''), which is pretty tedious.  You can also run into problems if the fields are different datatypes (string vs numeric), so sometimes you need to cast as well.
+You can get around this by wrapping each of your columns in a `coalesce` function to default nulls to an alternate value, which is pretty tedious.  You can also run into problems if the fields are different datatypes (string vs numeric), so sometimes you need to cast as well.
 
 ```sql
 ...
 select
   *,
   concat(
-    coalesce(cast(user_id as string), ''),
-    coalesce(cast(product_id as string), '')
+    coalesce(cast(user_id as string), '_this_used_to_be_null_'),
+    coalesce(cast(product_id as string), '_this_used_to_be_null_')
+    ) as _surrogate_key
+from example_ids
+```
+output:
+
+| USER_ID | PRODUCT_ID | _SURROGATE_KEY |
+|---------|------------|----------------|
+| 123     | 123        | 123123         |
+| 123     | `null`     | 123_this_used_to_be_null_            |
+| `null`  | 123        | _this_used_to_be_null_123            |
+
+Much better! But let's add another row to our dataset: 
+```sql
+with 
+
+example_ids as (
+  select
+  123 as user_id,
+  123 as product_id
+
+  union all
+
+  select
+  123 as user_id,
+  null as product_id
+
+  union all
+
+  select
+  null as user_id,
+  123 as product_id
+
+  union all 
+  
+  select 1231 as user_id,
+  23 as product_id
```

Comment:
Had to add this because the changes make it harder to wind up in a bad state (yay!)